dso/util.py

```diff
@@ -1,7 +1,55 @@
 import functools
 import logging
+import uuid
+
+import dso.model
+import dso.compliancedb.db
+import dso.compliancedb.model
+
+
+logger = logging.getLogger(__name__)
 
 
 @functools.lru_cache
 def component_logger(name: str):
     return logging.getLogger(name)
+
+
+def scans_to_compliance_db(
+    scan_data,
+    tool: dso.compliancedb.model.ScanTool,
+    component_name: str,
+    component_version: str,
+    artifact_name: str,
+    artifact_version: str,
+    artifact_type: dso.model.ArtifactType,
+    compliancedb_cfg_name: str,
+):
+    if not compliancedb_cfg_name:
+        logger.warning('no compliance db cfg name found, skipping insertion')
+        return
+    logger.info('inserting results to database')
+
+    db_fac = dso.compliancedb.db.ComplianceDBFactory(cfg_name=compliancedb_cfg_name)
+    cdb = db_fac.make()
+
+    # TODO ensure uuids are actually unique
```

Comment:
haha.. (https://en.wikipedia.org/wiki/Universally_unique_identifier#Version_4_(random)). no need to be _that_ zealous ^^)

however, it might be better to use a different means to identify artifacts, like so:

```
component_id = f'{componentName/componentVersion}'
artifact_id = str(artifact.identity())

qualified_artifact_id = '/'.join(component_id, artifact_id)
```

Those IDs may become a bit lengthy, so we _might_ consider storing a hash-digest, rather than the full str (`hashlib.sha256(qualified_artifact_id).hexdigest()`)

Reply:
Yea, sure thing, but we have to discuss what is considered as `artifact.identity`.